scripts/coverage_report_jenkins.py

```diff
@@ -485,6 +606,93 @@ def main(appliance, jenkins_url, jenkins_user, jenkins_token, job_name):
         merge_coverage_data(
             ssh=ssh,
             coverage_dir=coverage_dir)
+
+        # Increment index and wave count
+        i += wave_size
+        wave += 1
+
+        # We have to cleanup the coverage data we just extracted, move
+        # the merged .resultset.json file, and remove the merged data directory.
+        # We move the .resultset.json files so that it will be seen as just another
+        # result set to merge in the next wave, and the merged data directory is removed
+        # so coverage_merger.rb won't get confused by results already being where it drops
+        # it's results.   However it is important that we don't do that
+        # on the last wave, as we want the merge results to be available after the
+        # last wave.
+        #
+        # XXX: Yes, this is a hack.  Not even one I am proud of.
+        if i < len(builds):
```

Comment:
Could you structure this to run at the start of the loop, cleaning up the working area before download/extract?  So `cleanup_coverage_data_wave()` would return cleanly if its the first wave and there's nothing to clean up?  Less hackish I guess?

Reply:
Feels the same.   The hackish part is the whole cleaning up after the wave, not that I do the i < len(builds) check.   Having the cleanup at the end also feels natural (read expected), whereas having it at the beggining of the loop causes my eyebrows to raise.